pyproject.toml

```diff
@@ -142,3 +191,111 @@ reportUnusedParameter = false
 reportAny = false
 reportImplicitOverride = false
 reportAttributeAccessIssue = false  # adding our own private property on click commands
+
+# New Hatch environment configurations below
+
+[[tool.hatch.envs.test.matrix]]
+python = ["3.11", "3.12"]
+
+[tool.hatch.envs.test]
+description = "Run pytest with coverage across multiple Python versions."
+dependencies = [
+  "fromager[test]",  # Pulls in pytest, coverage, etc.
+]
+features = ["coverage"] # Enable Hatch's built-in coverage support
+scripts.run = "pytest --log-level DEBUG -vv {args:tests}"
+# To run: hatch test
+# To run with arguments: hatch test -- <pytest_args>
+# Coverage data will be automatically collected. Run `hatch coverage` for report.
+
+[tool.hatch.envs.lint]
+description = "Run linters and formatters checks."
+detached = true # Does not need a matrix, uses default python
+dependencies = [
+    "ruff",
+    "packaging", # As per tox.ini, potentially for mergify_lint.py
+    "PyYAML",    # As per tox.ini, potentially for mergify_lint.py
+]
+scripts.check = [
+    "ruff check src tests",
+    "ruff format --check src tests",
+    "python ./e2e/mergify_lint.py",
+]
+# To run: hatch run lint:check
+
+[tool.hatch.envs.fix]
+description = "Apply auto-fixes for linters and formatters."
+detached = true
+dependencies = ["ruff"]
+scripts.apply = [
+    "ruff format src tests docs",
+    "ruff check --fix src tests docs",
+]
+# To run: hatch run fix:apply
```

Comment:
We could use the default env for running the cli, so we don't need this.

Reply:
fixed it